src/app/pages/home/components/home-tabs.tsx

```diff
@@ -13,53 +14,54 @@ interface HomeTabsProps extends StackProps {
   activity: React.JSX.Element;
 }
 
-const ANALYTICS_PATH = ['/balances', '/activity'];
+export function HomeTabs({ balances, activity }: HomeTabsProps) {
+  const navigate = useNavigate();
+  const { pathname } = useLocation();
 
-export function HomeTabs(props: HomeTabsProps) {
-  const { balances, activity, ...rest } = props;
-  const analytics = useAnalytics();
+  const tabs = useMemo(
+    () => [
+      { slug: RouteUrls.Home, label: 'Balances' },
+      { slug: RouteUrls.Activities, label: 'Activity' },
+    ],
+    []
+  );
 
-  const { activeTab, setActiveTab } = useHomeTabs();
+  const getActiveTab = useCallback(
+    () => tabs.findIndex(tab => tab.slug === pathname),
+    [tabs, pathname]
+  );
 
-  const setActiveTabTracked = (index: number) => {
-    void analytics.page('view', ANALYTICS_PATH[index]);
-    setActiveTab(index);
-  };
+  const setActiveTab = useCallback(
+    (index: number) => navigate(tabs[index]?.slug),
+    [navigate, tabs]
+  );
 
   return (
-    <Stack flexGrow={1} mt="loose" spacing="extra-loose" {...rest}>
+    <Stack flexGrow={1} mt="loose" spacing="extra-loose">
       <Tabs
-        tabs={[
-          { slug: 'balances', label: 'Balances' },
-          { slug: 'activity', label: 'Activity' },
-        ]}
-        activeTab={activeTab}
-        onTabClick={setActiveTabTracked}
+        tabs={tabs}
+        activeTab={getActiveTab()}
+        onTabClick={setActiveTab}
         width={['100%', '193px']}
       />
       <Flex position="relative" flexGrow={1}>
-        {activeTab === 0 && (
-          <Suspense fallback={<LoadingSpinner pb="72px" />}>
-            <SlideFade in={activeTab === 0}>
-              {styles => (
-                <Box style={styles} width="100%">
-                  {balances}
-                </Box>
-              )}
-            </SlideFade>
-          </Suspense>
-        )}
-        {activeTab === 1 && (
-          <Suspense fallback={<LoadingSpinner pb="72px" />}>
-            <SlideFade in={activeTab === 1}>
-              {styles => (
-                <Box width="100%" style={styles}>
-                  {activity}
-                </Box>
-              )}
-            </SlideFade>
-          </Suspense>
-        )}
+        <Suspense fallback={<LoadingSpinner pb="72px" />}>
+          <SlideFade in={true}>
+            {styles => (
+              <Box style={styles} width="100%">
```

Comment:
Don't fully understand what's going on here, but wonder if there's a way to achieve this routing set up with `<Outlet />` where the routing structure where we describe the activity component on the route?

`<Route path={RouteUrls.Activities} element={<ActivityList />} />`?  

Home already has an outlet, and this might require you to move it to the tabs, but think that's okay? Worth trying to do it this way as I think that'd be more to the spec,  but if too difficult, this implementation is still an improvement.

Reply:
Thanks @kyranjamie I'll check that out. I was trying to use sub-routes in home tabs but I was facing an issue where it kept re-directing back to `/`. 

I'll try again and also move the outlet and get rid of that warning, thanks!